system_test.sh

```diff
@@ -22,7 +22,16 @@ function try { let tests+=1; this="$1"; }
 
 function assert {
 	[[ "$1" == "$2" ]] && { printf "."; return; }
-	printf "\nFAIL: $this\n'$1' != '$2'\n"; exit 1
+	printf "\nFAIL: $this\n'$1' != '$2'\n"
+
+	echo "*************************************"
+	echo "Preceding stdout:"
+	cat $tmp/exec.out
+	echo "*************************************"
+	echo "Preceding stderr:"
+  cat $tmp/exec.err
+	echo "*************************************"
+  exit 1
```

Comment:
It would be even better if printed the names and contents of all of the files in the temporary directory. Something like this:

```bash
for f in `find . -type f`; do
  echo "== $f =="; cat $f
done
```

Reply:
Catting everything might result in too much output. How about the new version I just pushed that limits output to 50 lines? It should be enough to understand what the files are and what happened in the test.

Reply:
There will very little output (< 12 lines) since the temproary directory is cleared in the setup of each test

Reply:
I see; `head` will print the name of each file